src/components/MainPage.js

```diff
@@ -0,0 +1,188 @@
+import * as React from "react";
+import AppBar from "@mui/material/AppBar";
+import Button from "@mui/material/Button";
+import CameraIcon from "@mui/icons-material/PhotoCamera";
+import Card from "@mui/material/Card";
+import CardActions from "@mui/material/CardActions";
+import CardContent from "@mui/material/CardContent";
+import CardMedia from "@mui/material/CardMedia";
+import CssBaseline from "@mui/material/CssBaseline";
+import Grid from "@mui/material/Grid";
+import Box from "@mui/material/Box";
+import Typography from "@mui/material/Typography";
+import Container from "@mui/material/Container";
+import { createTheme, ThemeProvider } from "@mui/material/styles";
+import Labels from "./Labels";
+import { Chip } from "@mui/material";
+import Progress from "./Progress";
+import reactJsLogo from "../logoImages/react-logo.png";
+import jsLogo from "../logoImages/js.svg";
+import dataStructureLogo from "../logoImages/data-structure.png";
+
+const quizCards = [
+  {
```

Comment:
Same comment about the ID-as-object-keys structure that I proposed for the userProgress object applies here.

Reply:
I'm not sure what this comment means, can you please clarify?